File: Python/Greedy Algorithms/L_maximumIcecreamBar.py
from typing import List
import heapq

def maxIceCream(costs: List[int], coins: int) -> int:
        heapq.heapify(costs)
        total = 0
        while costs and coins:
            price = heapq.heappop(costs)
            if price > coins:
                return total
            coins -= price
            total += 1
        return total

File: cpp/Greedy/L_maximumIcecreamBar.cpp
#include<bits/stdc++.h>
using namespace std;

int maxIceCream(vector<int>& costs, int coins) {
    sort(costs.begin(), costs.end());
    int total = 0;
    for(int price: costs) {
        if (price > coins) return total;
        coins -= price;
        total++;
    }
    return total;
}